internal/storage/v1/elasticsearch/spanstore/reader.go

```diff
@@ -677,6 +681,75 @@ func (*SpanReader) buildObjectQuery(field string, k string, v string) elastic.Qu
 	return elastic.NewBoolQuery().Must(keyQuery)
 }
 
+func (s *SpanReader) mergeAllNestedAndElevatedTagsOfSpan(span *dbmodel.Span) error {
+	processTags, err := s.mergeNestedAndElevatedTags(span.Process.Tags, span.Process.Tag)
+	if err != nil {
+		return err
+	}
+	span.Process.Tags = processTags
+	spanTags, err := s.mergeNestedAndElevatedTags(span.Tags, span.Tag)
+	if err != nil {
+		return err
+	}
+	span.Tags = spanTags
+	return nil
+}
+
+func (s *SpanReader) mergeNestedAndElevatedTags(nestedTags []dbmodel.KeyValue, elevatedTags map[string]any) ([]dbmodel.KeyValue, error) {
+	mergedTags := make([]dbmodel.KeyValue, len(nestedTags)+len(elevatedTags))
+	copy(mergedTags[:len(nestedTags)], nestedTags)
```

Comment:
It's not necessary to fiddle with indices
```suggestion
	mergedTags := make([]dbmodel.KeyValue, 0, len(nestedTags)+len(elevatedTags))
	mergedTags = append(mergedTags, nestedTags...)
```